package.json

```
@@ -3,37 +3,53 @@
   "version": "7.0.0-beta.0",
   "description": "The router for easy microfrontends",
   "type": "module",
-  "main": "lib/umd/single-spa.min.cjs",
+  "main": "./lib/umd/single-spa.min.cjs",
+  "types": "./lib/umd/types/single-spa.d.cts",
+  "module": "./lib/esm/single-spa.min.js",
   "publishConfig": {
     "access": "public"
   },
   "exports": {
     ".": {
       "development": {
-        "import": "./lib/esm/single-spa.dev.js",
-        "require": "./lib/umd/single-spa.dev.cjs",
-        "types": "./typings/single-spa.d.ts"
+        "import": {
+          "types": "./lib/esm/types/single-spa.d.ts",
+          "default": "./lib/esm/single-spa.min.js"
+        },
+        "require": {
+          "types": "./lib/umd/types/single-spa.d.cts",
+          "default": "./lib/umd/single-spa.min.cjs"
+        }
       },
       "production": {
-        "import": "./lib/esm/single-spa.min.js",
-        "require": "./lib/umd/single-spa.min.cjs",
-        "types": "./typings/single-spa.d.ts"
+        "import": {
+          "types": "./lib/esm/types/single-spa.d.ts",
+          "default": "./lib/esm/single-spa.min.js"
+        },
+        "require": {
+          "types": "./lib/umd/types/single-spa.d.cts",
+          "default": "./lib/umd/single-spa.min.cjs"
+        }
       },
       "default": {
-        "import": "./lib/esm/single-spa.min.js",
-        "require": "./lib/umd/single-spa.min.cjs",
-        "types": "./typings/single-spa.d.ts"
+        "import": {
+          "types": "./lib/esm/types/single-spa.d.ts",
+          "default": "./lib/esm/single-spa.min.js"
+        },
+        "require": {
+          "types": "./lib/umd/types/single-spa.d.cts",
+          "default": "./lib/umd/single-spa.min.cjs"
+        }
       }
     },
     "./package.json": "./package.json",
     "./lib/*": "./lib/*"
   },
   "scripts": {
-    "build": "pnpm run clean && concurrently pnpm:build:dev pnpm:build:prod pnpm:build:types",
+    "build": "pnpm run clean && concurrently pnpm:build:dev pnpm:build:prod && node ./scripts/remove-extraneous-types.cjs",
     "build:prod": "rollup -c --environment ROLLUP_ENV:'production'",
     "build:dev": "rollup -c",
     "build:analyze": "rollup -c --environment ANALYZER:'analyzer'",
-    "build:types": "tsc",
     "watch": "rollup -c -w",
     "clean": "rimraf lib typings",
```

Now that the `typings` folder isn't being created during the build, it should be removed from the `clean` script as well as `.gitignore`, `.prettierignore`, and `tsconfig.json`

```suggestion
    "clean": "rimraf lib",
```

---

Done.